src/main/js/dygraph-light/main/LabelMaker.ts

```diff
@@ -200,25 +199,26 @@ class Metadata {
 	}
 }
 
-const getFinalLabel = (linking: 'overlap' | 'concatenate', requestedLabels: string[], lblIdx: number, object: MetaWithTableFormat, colName: string, postfix: string = '') => {
-	if (linking === 'concatenate'){
-		return (requestedLabels.length > lblIdx && requestedLabels[lblIdx].length
-			? requestedLabels[lblIdx]
-			: getLabel(object.tableFormat, colName)) + postfix;
+const getFinalLabel = (linking: 'overlap' | 'concatenate', requestedLabel: string, object: MetaWithTableFormat, colName: string, postfix: string = '') => {
+	if (requestedLabel) return requestedLabel + postfix;
 
-	} else {
-		const filename = object.filename;
+	if (linking === 'concatenate'){
+		return getLabel(object.tableFormat, colName) + postfix;
+	} 
+	const filename = object.filename;
 
-		return (requestedLabels.length > lblIdx && requestedLabels[lblIdx].length
-			? requestedLabels[lblIdx]
-			: filename.slice(0, filename.lastIndexOf('.'))) + postfix;
-	}
+	return filename.slice(0, filename.lastIndexOf('.')) + postfix;
 };
 
-const getLegendLabelsFromParams = (params: UrlSearchParams, name: string) => {
-	return params.has(name)
-		? params.get(name).split(',') as string[]
-		: [];
+const getLegendLabelFromParams = (params: UrlSearchParams, name: string, objId: string) => {
```

Comment:
There's a practice I picked up at some point and try to follow, about avoiding multiple arguments of the same type in function signatures, since it significantly reduces accidentally mixing things up at call sites.
It's of course not something to follow religiously, as with everything, and becomes more important the more arguments there are and the larger the scope access scope of the function is.
In this case it's maybe overkill to introduce a type for an "Object descriptor", but maybe this would be a low-overhead improvement?
```suggestion
const getLegendLabelFromParams = (params: UrlSearchParams, {name, objId}: {name: string, objId: string}) => {
```